### PR TITLE
unicode errors

### PR DESCRIPTION
**Pull request by s2hc-johan**
make decode if we're using python2

### CODE REVIEW

File: v7/recent_posts_json/recent_posts_json.py [hunk 1]
```diff
@@ -131,7 +132,10 @@ def make_json(self, posts, descriptions, previewimage, output_path, lang):
             recent_posts.append(entry)
         data = json.dumps(recent_posts, indent=2, sort_keys=True)
         with io.open(output_path, "w+", encoding="utf8") as outf:
-            outf.write(data)
+            if sys.version_info[0] != 3:
```
Comment (Kwpolska) [inline v7/recent_posts_json/recent_posts_json.py]: This is not right.

On Python 2, json.dumps() might return Unicode in some cases. It doesn’t by default; yet we could use to change the default:

```py
data = json.dumps(recent_posts, ensure_ascii=False, indent=2, sort_keys=True)
with io.open(output_path, "w+", encoding="utf-8") as outf:
    try:
        outf.write(data.decode('utf-8'))
    except AttributeError:
        outf.write(data)
````

Comment (s2hc-johan) [inline v7/recent_posts_json/recent_posts_json.py]: Absolutley we can do it like that. Don't know why the first commit is wrong though, isn't json utf-8 by design? In ptyhon2 ".decode('utf-8')" works on both string and unicode objects

Comment (Kwpolska) [inline v7/recent_posts_json/recent_posts_json.py]: It does not work properly.

```pytb
>>> u"ą".decode('utf-8')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeEncodeError: 'ascii' codec can't encode character u'\u0105' in position 0: ordinal not in range(128)
```

Please switch it to the solution I recommended.

Comment (Kwpolska) [inline v7/recent_posts_json/recent_posts_json.py]: Actually, in this case we need

```py
except (AttributeError, UnicodeEncodeError, UnicodeDecodeError):

Comment (s2hc-johan) [inline v7/recent_posts_json/recent_posts_json.py]: yea, excepting more thing makes it more clear.

We don't decode random unicode, we decode output from json.dumps:
    
    >>> import json
    >>> json.dumps([u"ą"]).decode('utf-8')
    u'["\\u0105"]'
    >>>

Comment (Kwpolska) [inline v7/recent_posts_json/recent_posts_json.py]: Just do it with `ensure_ascii=False`.  More modern.

**Comment by ralsina** [comment]
We no longer care about python 2

**Pull request closed without merging**